libpcsxcore/new_dynarec/new_dynarec.c

```diff
@@ -3445,9 +3446,16 @@ void hlecall_assemble(int i,struct regstat *i_regs)
   assert(!is_delayslot);
   (void)ccreg;
   emit_movimm(start+i*4+4,0); // Get PC
-  emit_movimm((int)psxHLEt[source[i]&7],1);
-  emit_addimm(HOST_CCREG,CLOCK_ADJUST(ccadj[i]),HOST_CCREG); // XXX
-  emit_jmp((int)jump_hlecall);
+  uint32_t hleCode = psxRegs.code & 0x03ffffff;
```

Comment:
You can't use `psxRegs.code` here as it's never updated by the dynarec. Just use source[i].

Also the block must end with emit_jmp(), else the generated code will crash. So it should look something like this:
```
  uint32_t hleCode = source[i] & 0x03ffffff;
  if (hleCode >= (sizeof(psxHLEt) / sizeof(psxHLEt[0])))
    emit_movimm((int)psxNULL,1);
  else
    emit_movimm((int)psxHLEt[hleCode],1);
  emit_addimm(HOST_CCREG,CLOCK_ADJUST(ccadj[i]),HOST_CCREG); // XXX
  emit_jmp((int)jump_hlecall);
```

Also this file doesn't have a single tab, it would be great if you could avoid adding them as they look different on different editors and things get misaligned in an ugly way.